config/file_config.go

```diff
@@ -30,6 +32,54 @@ func (d *Duration) UnmarshalText(text []byte) error {
 	return nil
 }
 
+const (
+	K = uint64(1000)
+	M = 1000 * K
+	G = 1000 * M
+	T = 1000 * G
+	P = 1000 * T
+	E = 1000 * P
+
+	Ki = uint64(1024)
+	Mi = 1024 * Ki
+	Gi = 1024 * Mi
+	Ti = 1024 * Gi
+	Pi = 1024 * Ti
+	Ei = 1024 * Pi
+
+	invalidSizeError = "invalid size: %s"
+)
+
+var binaryMap = map[string]uint64{
```

Comment:
this isn't wrong, but why do you have two maps instead of one? The reason to have two is if you need to support one and not the other, but we don't have that need. Unifying these two maps would give you a single lookup and simplify the logic.